browser/src/control/Control.UIManager.js

```diff
@@ -1022,21 +1022,30 @@ L.Control.UIManager = L.Control.extend({
 	/// Shows tooltip in the document area
 	/// tooltipInfo contains rectangle (position in twips) and text properties
 	/// elem has to be jQuery selector output eg. $('.leaflet-layer')
-	showDocumentTooltip: function(tooltipInfo, elem) {
-		var split = tooltipInfo.rectangle.split(',');
-		var latlng = this.map._docLayer._twipsToLatLng(new L.Point(+split[0], +split[1]));
-		var pt = this.map.latLngToContainerPoint(latlng);
-
-		elem.tooltip();
-		elem.tooltip('enable');
-		elem.tooltip('option', 'content', tooltipInfo.text);
-		elem.tooltip('option', 'items', elem[0]);
-		elem.tooltip('option', 'position', { my: 'left bottom',  at: 'left+' + pt.x + ' top+' + pt.y, collision: 'fit fit' });
-		elem.tooltip('open');
-		document.addEventListener('mousemove', function() {
-			elem.tooltip('close');
-			elem.tooltip('disable');
-		}, {once: true});
+	showDocumentTooltip: function(tooltipInfo, elem, pos) {
+		var pt = this.map.latLngToContainerPoint(pos);
+		pt.y -=35; //Show tooltip above the cursor.
+
+		if ($('.ui-tooltip').length > 0) {
+			elem.tooltip('option', 'content', tooltipInfo);
+		}
+		else {
+			elem.tooltip({
+				content: tooltipInfo,
+				items: elem[0],
+                position: { my: 'left top', at: 'left+' + pt.x +  ' top+' +pt.y, collision: 'fit fit' }
+			});
+			elem.tooltip('open');
+			elem.off('mouseleave');
+		}
+
+		document.addEventListener('keydown', function closeTooltip(e) {
+			if (e.key === 'Escape') {
+				elem.tooltip('close');
+				elem.tooltip('disable');
+				document.removeEventListener('keydown', closeTooltip);
```

Comment:
removing callback can be done using { once: true } parameter, look in addEventListener documentation